### PR TITLE
Tweaking titles and links to support glTF schema title changes

### PR DESCRIPTION
**Pull request by BeardedGnome**
Two changes:
1) Formatted the properties to look like 'code' properties. The capitalization and spacing in the title wasn't appropriate here.
2) Created a special linker for the TOC that removes the parent type string from the current type string, displays the shorted string and links to the original one

These are to support the changes here: https://github.com/KhronosGroup/glTF/pull/1014

@bghgary
@HowardWolosky

### CODE REVIEW

**Comment by pjcozzi** [comment]
Sounds good.  @HowardWolosky please merge when ready.

@BeardedGnome do you want commit access to this repo so you can also merge pull requests?  I have not been actively contributing here recently, and this project could use the support.

**Comment by BeardedGnome** [comment]
I'm ramping down on glTF/Wetzel contributions. So giving me access is not necessary.

**Comment by pjcozzi** [comment]
Gotcha.

File: lib/generateMarkdown.js [hunk 1]
```diff
@@ -103,14 +103,26 @@ function getRecursiveTOC(orderedTypes, parentType, depth) {
...
-            md += style.bulletItem(style.linkType(style.typeValue(currentType), currentType, enums.autoLinkOption.codeQuoteOnly), depth);
+            md += style.bulletItem(getTOCLink(currentType.replace(parentType + " ", ""), currentType), depth);
             md += getRecursiveTOC(orderedTypes, currentType, depth + 1);
         }
     }
     return md;
 }
 
 /**
+* @function getTOCLink
```
Comment (HowardWolosky) [inline lib/generateMarkdown.js]: Markdown-specific logic (like forming links) should be in style.js.  And, I already had bad form by having two separate functions hardcode the `reference` string, and now you're adding a third.  Can you make a constant within style.js for that, and then use it in the two existing functions as well as your new one?

File: lib/generateMarkdown.js [hunk 2]
```diff
@@ -103,14 +103,26 @@ function getRecursiveTOC(orderedTypes, parentType, depth) {
...
+* @param  {string} displayString The text to display in the link.
+* @param  {string} type          The string to link to.
+* @return {string} The markdown for a link with displayString text targeted at type.
+*/
+function getTOCLink(displayString, type)
+{
+    var typeLink = '#reference-' + type.toLowerCase().replace(/ /g, '-');
+    return "[" + style.typeValue(displayString) + '](' + typeLink + ")";
```
Comment (HowardWolosky) [inline lib/generateMarkdown.js]: We already have a method that knows how to make MD links: `style.getLinkMarkdown`.  Please use that.

Comment (BeardedGnome) [inline lib/generateMarkdown.js]: Is there a reason style.linkType doesn't use style.getLinkMarkdown?

Comment (HowardWolosky) [inline lib/generateMarkdown.js]: No.  It should have.  That was an oversight that wasn't caught previously.

**Comment by BeardedGnome** [comment]
Updated PR. Current output: https://gist.github.com/BeardedGnome/55077c3f74b97137712f32b8ad0f9da1

**Comment by BeardedGnome** [comment]
`orthographic` is a variable name of type `object`. The fact that there used to be a data type also called orthographic was a pure coincidence. In fact, we often linked to the wrong type when we had a variable and type with the same name (sampler vs. [animation] sampler). This is a big issue with the way glTF mixes and matches variable names and type names, not something that Wetzel should try to fix.

**Comment by HowardWolosky** [comment]
Alright, let's close this one out.  This is doing the right thing for the TOC.  For the time being, it seems that if people want more correct auto-linking for glTF, they'll need to use the proper type names in code-quotes within the description and then use the `cqo` option for auto-linking,.

**Comment by HowardWolosky** [comment]
Thanks for the change!

**Pull request merged**